packages/xstate-solid/src/utils.ts

```diff
@@ -0,0 +1,61 @@
+function isObjectOrArray(value: any): value is object {
+  return value && typeof value === 'object';
+}
+
+function isNonObjectPrototype(value: any): value is object {
+  return (
+    value?.constructor?.name &&
+    value.constructor.name !== 'Object' &&
+    value.constructor.name !== 'Array'
+  );
+}
+
+/**
+ * Accepts any value and creates a deep clone if it is an object
+ * This function only deeply clones objects, any classes with be copied
+ * @param value The variable to deeply clone
+ * @param valueRefs An array that stores a reference to the properties in the original object
+ * @param clonedValueRefs An array that stores a reference to cloned
+ */
+const clone = <T>(
+  value: T,
+  valueRefs: unknown[],
+  clonedValueRefs: unknown[]
+): T => {
+  if (!isObjectOrArray(value)) {
+    return value;
+  }
+
+  // Save a reference of the object/array
+  valueRefs.push(value);
+
+  // If the value is a class or non object/array prototype, copy over instead of cloning
+  if (isNonObjectPrototype(value)) {
```

Comment:
We only need to deeply clone object & arrays. Classes and other prototypes are not altered by wrapping from Solid stores. I have a couple tests to verify the same behavior between this integration and vanilla Solid stores.